### PR TITLE
Added support for implementing v2 API calls

### PR DESCRIPTION
**Pull request by bartbroere**
Added support for v2 API calls
The commit is backwards compatible
New API methods now raise the same type of error

### CODE REVIEW

File: bitstamp/client.py [hunk 1]
```diff
@@ -57,12 +57,11 @@ def _construct_url(self, url, base, quote):
         """
         Adds the orderbook to the url if base and quote are specified.
         """
-        #This condition allows for both None and False values
-        if not base and not quote:
+        if (base is None) and (quote is None):
```
Comment (kmadac) [inline bitstamp/client.py]: Previous condition was more Pythonic. Why you changed it this way?

File: bitstamp/client.py [hunk 2]
```diff
@@ -57,12 +57,11 @@ def _construct_url(self, url, base, quote):
...
             return url
         else:
-            #The join method is quicker than standard concatenation
-            url.join([base.lower(), quote.lower(), "/"])
+            #TODO: The join method is quicker than standard concatenation
```
Comment (kmadac) [inline bitstamp/client.py]: Do we need this TODO? Is speed optimization really necessary here?

File: bitstamp/client.py (hunk 1 above)
Comment (bartbroere) [inline bitstamp/client.py]: Thanks for the remark. There was an error in _construct_url and I wanted to make sure it was not that. I have restored the original.

File: bitstamp/client.py (hunk 2 above)
Comment (bartbroere) [inline bitstamp/client.py]: As far as I'm concerned, it is fine to do it with the +. I will remove the TODO.

**Comment by kmadac** [comment]
Hi @bartbroere, it's great that you are working on API v2 update. I will try to check commits you made constantly, but please let me know here once your work will be done and I will try PR as soon as possible. Of course we need to have unit tests made for all new functions.

**Comment by bartbroere** [comment]
Hi. No problem, I'm happy you're considering merging it. I have added placeholder unit tests, and I am testing updated methods with my own API key to make sure it all works as expected. If there's time I will make the unit tests tomorrow.

There is of course new functionality to the API client, so a new version number might be necessary when you merge the pull request. I will let you know when I'm done.

**Comment by bartbroere** [comment]
The new API methods did not raise errors in the same way as the old one. This now has been fixed. @kmadac I think the pull request is ready to be merged.

**Pull request merged**